File: owlify-famix/src/test/java/examples/ClassWithInnerClass.java
package examples;

public class ClassWithInnerClass {
    private InnerClass field1;
    private float field2;

    protected ClassWithInnerClass(int parameter1, float parameter2) {
        field1 = new InnerClass(parameter1);
        field2 = parameter2;
    }

    public double method() {
        return field1.get() * field2;
    }

    private static class InnerClass {
        private final int innerField;

        public InnerClass(int innerParameter) {
            innerField = innerParameter;
        }

        public int get() {
            return innerField;
        }
    }
}

File: kotlin-parser/src/test/resources/example-project/example/ClassWithInnerClass.kt
package example

class ClassWithInnerClass protected constructor(parameter1: Int, parameter2: Float) {
    private val field1: InnerClass
    private val field2: Float
    fun method(): Double {
        return (field1.get() * field2).toDouble()
    }

    private class InnerClass(private val innerField: Int) {
        fun get(): Int {
            return innerField
        }

        private class InsideClass(private val insideField: Int) {
            fun get(): Int {
                return insideField
            }
        }
    }

    init {
        field1 = InnerClass(parameter1)
        field2 = parameter2
    }
}